iroh-store/src/store.rs

```diff
@@ -291,6 +334,38 @@ impl Store {
         }
     }
 
+    fn get_ids_for_hash(
+        &self,
+        hash: &Multihash,
+    ) -> Result<impl Iterator<Item = Result<(u64, u64)>> + '_> {
```

Comment:
It is very unfortunate to return a tuple of `u64`s, maybe use a struct to avoid mixing them up? Or we could introduce a new type for `Id`.

Reply:
A newtype for Id will be quite annoying since it is used all over the place. But I will make one for Code so you can not accidentally use it as an id.